packages/react-components/react-tags-preview/stories/InteractionTag/InteractionTagDismiss.stories.tsx

```diff
@@ -8,37 +8,86 @@ import {
   { value: '2', children: 'Tag 2' },
   { value: '3', children: 'Tag 3' },
 ];
 
+/**
+ * focus management for the reset button
+ */
+const useResetExample = (visibleTagsLength: number) => {
+  const resetButtonRef = React.useRef<HTMLButtonElement>(null);
+  const firstTagRef = React.useRef<HTMLButtonElement>(null);
+
+  const prevVisibleTagsLengthRef = React.useRef<number>(visibleTagsLength);
+  React.useEffect(() => {
+    if (visibleTagsLength === 0) {
+      resetButtonRef.current?.focus();
+    } else if (prevVisibleTagsLengthRef.current === 0) {
+      firstTagRef.current?.focus();
+    }
+
+    prevVisibleTagsLengthRef.current = visibleTagsLength;
+  }, [visibleTagsLength]);
+
+  return { firstTagRef, resetButtonRef };
+};
+
 export const Dismiss = () => {
   const [visibleTags, setVisibleTags] = React.useState(initialTags);
   const removeItem: TagGroupProps['onDismiss'] = (_e, { dismissedTagValue }) => {
     setVisibleTags([...visibleTags].filter(tag => tag.value !== dismissedTagValue));
   };
+  const resetItems = () => setVisibleTags(initialTags);
+  const { firstTagRef, resetButtonRef } = useResetExample(visibleTags.length);
+
+  const styles = useStyles();
 
   return (
-    <TagGroup onDismiss={removeItem} aria-label="Dismiss example">
-      {visibleTags.map(tag => {
-        const primaryId = `dismiss-primary-${tag.value}`;
-        const secondaryId = `dismiss-secondary-${tag.value}`;
-        return (
-          <InteractionTag value={tag.value} key={tag.value}>
-            <InteractionTagPrimary id={primaryId} hasSecondaryAction>
-              {tag.children}
-            </InteractionTagPrimary>
-            <InteractionTagSecondary
-              id={secondaryId}
-              aria-label="remove"
-              aria-labelledby={`${primaryId} ${secondaryId}`}
-            />
-          </InteractionTag>
-        );
-      })}
-    </TagGroup>
+    <div className={styles.container}>
+      {visibleTags.length !== 0 && (
+        <TagGroup onDismiss={removeItem} aria-label="Dismiss example">
+          {visibleTags.map((tag, index) => {
+            const primaryId = `dismiss-primary-${tag.value}`;
+            const secondaryId = `dismiss-secondary-${tag.value}`;
+            return (
+              <InteractionTag value={tag.value} key={tag.value}>
+                <InteractionTagPrimary id={primaryId} hasSecondaryAction ref={index === 0 ? firstTagRef : null}>
+                  {tag.children}
+                </InteractionTagPrimary>
+                <InteractionTagSecondary
+                  id={secondaryId}
+                  aria-label="remove"
+                  aria-labelledby={`${primaryId} ${secondaryId}`}
+                />
+              </InteractionTag>
+            );
+          })}
+        </TagGroup>
+      )}
+
+      <Button
+        onClick={resetItems}
+        ref={resetButtonRef}
+        disabled={visibleTags.length !== 0}
+        className={styles.resetButton}
```

Comment:
```suggestion
        className={styles.resetButton}
        size="small"
```

nit: to make it clear that it's not a part of the example